pkg/util/ansible_execution.go

```diff
@@ -266,23 +266,20 @@ func getAnsibleExecutionNamePrefix(serviceName string) string {
 
 // GetAnsibleExecutionNameAndLabel Name and Label of AnsibleEE
 func GetAnsibleExecutionNameAndLabel(service *dataplanev1.OpenStackDataPlaneService,
-	deployment client.Object,
-	nodeset client.Object) (string, map[string]string) {
-	executionName := fmt.Sprintf("%s-%s", getAnsibleExecutionNamePrefix(service.Name), deployment.GetName())
+	deploymentName string,
+	nodeSetName string) (string, map[string]string) {
+	executionName := fmt.Sprintf("%s-%s", getAnsibleExecutionNamePrefix(service.Name), deploymentName)
 	if !service.Spec.DeployOnAllNodeSets {
-		executionName = fmt.Sprintf("%s-%s", executionName, nodeset.GetName())
+		executionName = fmt.Sprintf("%s-%s", executionName, nodeSetName)
 	}
 	if len(executionName) > AnsibleExcecutionNameLabelLen {
 		executionName = executionName[:AnsibleExcecutionNameLabelLen]
 	}
-	labelValue := string(deployment.GetUID())
```

Comment:
I had continued using uid for labels to ensure that long deployment and nodeset names don't make us truncate the labels for AEE. We could have used uid for AEE names too, but we assert the names in kuttl tests (and the plan was to change those). There could be still be name collision, if the service, deployment and nodeset  names are not kept small and unique.

Reply:
That's a good point. Labels need to follow [RFC-952](https://www.rfc-editor.org/rfc/rfc952), or used to, now the [documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) specifies something more like [RFC-1123](https://www.rfc-editor.org/rfc/rfc1123).

There are ways to handle that though, by setting validation for nodeset, deployment, and service name. As it happens, we already have a package for that.

Reply:
I've added validation for nodeset, deployment and service names.

Reply:
k8s object names can be 253 chars. By limit deployment/nodeset names to 63 chars would not achieve anything as we use a combination of deployment and nodeset name to build the label in this PR.

Reply:
Some of them do, yes. But the validation here is meant to sanitize strings we'll eventually use as labels. These are limited to 63 chars.